### DISCUSSION TITLE
Setting :expire_after on session cookies

### DISCUSSION THREAD

**Discussion by muellerj** [jeremyevans/roda#244] [General]
I'm having trouble getting a persistent session (still there after browser restart) to work. I'm not sure if this should not be possible, please advise if so.

Setting `:expire_after` inside the `:cookie_options` has for the `sessions` plugin currently has no effect. 

``` ruby
  plugin :sessions,
    key: "my.session",
    secret: SESSION_SECRET,
    cookie_options: {
      domain: "example.com",
      expire_after: 2592000      # this does not make its way to the `Set-Cookie` header
    }
```

`Rack::Sessions`  does the processing [here](https://github.com/rack/rack/blob/main/lib/rack/session/abstract/id.rb#L395), but Roda sets the cookie without using this interface. Adding the following line to [`Roda::RodaPlugins::Sessions::RequestMethods#persist_session`](https://github.com/jeremyevans/roda/blob/master/lib/roda/plugins/sessions.rb#L269) does the job:

``` ruby
#  ...

elsif cookie_value = _serialize_session(session)
  cookie = Hash[opts[:cookie_options]]
  cookie[:value] = cookie_value
  cookie[:expires] = Time.now + cookie[:expire_after] if cookie[:expire_after]   # new
  cookie[:secure] = true if !cookie.has_key?(:secure) && ssl?
  Rack::Utils.set_cookie_header!(headers, opts[:key], cookie)
 end

# ...
```

Is this sound? If not, are there alternatives to be explored?

Kind regards,
Jonas

**Comment by jeremyevans** [comment]
Roda doesn't use `:expire_after`?  Rack's cookie handling recognizes the `:expires` and `:max_age` options.  Looks like Rack's session support will recognize `:expire_after` session option and set the `:expires` cookie option, but Roda's sessions plugin does not use Rack's session support.  You should probably use the `:max_age` cookie option.

**Comment by muellerj** [comment]
Awesome, thank you 👍
